### ISSUE TITLE
Many callbacks are not unregistered

### ISSUE THREAD

**Issue by xi** [liqd/adhocracy3.mercator#674]
We register callbacks in many places and only sometimes unregister them later on. As an example, `adhTopLevelState` doesn't even expose the functionality to unregister callbacks registered with `on()`.

I am not sure how big this problem is. It may either cause bugs (callback is triggered but referenced context does no longer exist) or prevent large amounts of data from being garbage collected.

Some contexts in which this might be relevant:

-  `adhTopLevelState` (`on()` and ` bind()`)
-  `adhWebSocket`
-  any instance of `EventHandler`

Basically a directive should unregister all callbacks on `$destroy`.

Further investigation needed.

**Comment by xi** [comment]
For reference: angular's [`$watch`](https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$watch) returns a function that, when called, unregisteres the callback. This could be a nice interface to make this simple.
